### PR TITLE
chore: set up snyk, fix vulnerabilities

### PR DESCRIPTION
**Pull request by aaron-steinfeld**
## Description
Setup snyk, update dependencies. Depends on https://github.com/hypertrace/kafka-streams-framework/pull/38

### CODE REVIEW

File: hypertrace-alert-engine/build.gradle.kts [hunk 1]
```diff
@@ -29,10 +29,10 @@ dependencies {
 
     // Logging
     implementation("org.slf4j:slf4j-api:1.7.30")
-    runtimeOnly("org.apache.logging.log4j:log4j-slf4j-impl:2.14.1")
+    runtimeOnly("org.apache.logging.log4j:log4j-slf4j-impl:2.15.0")
 
     // framework + libs
-    implementation("com.google.protobuf:protobuf-java-util:4.0.0-rc-2")
+    implementation("com.google.protobuf:protobuf-java-util:3.17.3")
```
Comment (aaron-steinfeld) [inline hypertrace-alert-engine/build.gradle.kts:35]: This one's actually a downgrade. Assuming all the tests are passing I'm going to take that as OK. We shouldn't be using prerelease artifacts, and we should be aligning with the version of protoc we're using elsewhere.

Comment (rish691) [reply hypertrace-alert-engine/build.gradle.kts:35]: @kotharironak was there any specific reason behind using this version?

File: .github/workflows/pr-build.yml [hunk 2]
```diff
@@ -40,4 +40,20 @@ jobs:
       - name: Build with Gradle
         uses: hypertrace/github-actions/gradle@main
         with: 
-          args: build -x avroCompatibilityCheck dockerBuildImages
\ No newline at end of file
+          args: build -x avroCompatibilityCheck dockerBuildImages
```
Comment (rish691) [inline .github/workflows/pr-build.yml:43]: I think we are publishing the avro classes, so avroCompatibility check will pass.

**Review by rish691 — APPROVED** [review]

File: hypertrace-alert-engine/build.gradle.kts (hunk 1 above)
Comment (kotharironak) [reply hypertrace-alert-engine/build.gradle.kts:35]: No. I don't think, there was any specific reason. I might have chosen directly from maven with latest one - https://mvnrepository.com/artifact/com.google.protobuf/protobuf-java

This is fine.

**Comment by rish691** [comment]
Merging this one to upgrade alerting in ht.

**Pull request merged**